### ISSUE TITLE
GDAL:  Building the GDAL results in clang hanging / crashing, failing build and taking down system

### ISSUE THREAD

**Issue by ArtificialRaccoon** [rhardih/bad#18]
(Note:  This wasn't happening previously / a few months ago)

If you attempt to do a "make gdal", the build will proceed up until a point.  This appears to be happening during the jpeg-install-obj step.  The recipe will fail to build for some reason, and clang will attempt to kill the job, but fail:

'''
make[1]: *** [jpeg-install-obj] Error 2
make[1]: *** Waiting for unfinished jobs
clang70++: error: unable to execute command: Killed
clang70++: error: clang frontend command failed due to signal
Android (4751641 based on r328903) clang version 7.0.2 ...
'''

After that, the host system becomes unresponsive and I'm forced to power down / restart.  So far it's happened on two separate systems.  

Is this happening for you?  Host systems have both been running Debian 11 for reference.

**Comment by ArtificialRaccoon** [comment]
I determined this isn't localized to bad, but to the build of the GDAL in general.  Running make with -j allows clang to gobble every bit of system resources until it kills the system.  Limiting the number of jobs prevents this from happening.

**Comment by rhardih** [comment]
I streamlined on just running with `-j` in 633275b6e75811a402a45b13817f7f6823036f43, but maybe that was a bad idea and instead it should be `-j 4` for all builds. 🤔 

I can't reproduce the exact issue you're having, but I'm able to exhaust the system, (16G MacBook Pro), to the point where I have to kill docker and start over.

It might not be crashing my system, because I've limited Docker globally to four cores and 8 gigs of memory.

<img width="478" alt="Screenshot 2023-01-22 at 10 17 49" src="https://user-images.githubusercontent.com/124250/213908937-b40bd251-6210-40b4-ac8f-ee17b5de085d.png">
